rioxarray/rioxarray.py

```diff
@@ -1051,6 +1051,105 @@ def slice_xy(self, minx, miny, maxx, maxy):
         subset.attrs["transform"] = tuple(self.transform(recalc=True))
         return subset
 
+    def pad_xy(self, minx, miny, maxx, maxy):
+        """Pad the array to x,y bounds.
+
+        .. versionadded:: 0.0.29
+
+        Parameters
+        ----------
+        minx: float
+            Minimum bound for x coordinate.
+        miny: float
+            Minimum bound for y coordinate.
+        maxx: float
+            Maximum bound for x coordinate.
+        maxy: float
+            Maximum bound for y coordinate.
+
+
+        Returns
+        -------
+        DataArray: A padded :class:`xarray.DataArray` object.
+
+        """
+        left, bottom, right, top = self._internal_bounds()
+        resolution_x, resolution_y = self.resolution()
+        y_before = y_after = 0
+        x_before = x_after = 0
+        y_coord = self._obj[self.y_dim]
+        x_coord = self._obj[self.x_dim]
+
+        if top - resolution_y < maxy:
+            new_y_coord = np.arange(bottom, maxy, -resolution_y)[::-1]
+            y_before = len(new_y_coord) - len(y_coord)
+            y_coord = new_y_coord
+            top = y_coord[0]
+        if bottom + resolution_y > miny:
+            new_y_coord = np.arange(top, miny, resolution_y)
+            y_after = len(new_y_coord) - len(y_coord)
+            y_coord = new_y_coord
+            bottom = y_coord[-1]
+
+        if left - resolution_x > minx:
+            new_x_coord = np.arange(right, minx, -resolution_x)[::-1]
+            x_before = len(new_x_coord) - len(x_coord)
+            x_coord = new_x_coord
+            left = x_coord[0]
+        if right + resolution_x < maxx:
+            new_x_coord = np.arange(left, maxx, resolution_x)
+            x_after = len(new_x_coord) - len(x_coord)
+            x_coord = new_x_coord
+            right = x_coord[-1]
+
+        superset = self._obj.pad(
+            pad_width={
+                self.x_dim: (x_before, x_after),
+                self.y_dim: (y_before, y_after),
+            },
+            constant_values=np.nan,
```

Comment:
Thoughts on attempting to get the nodata value to use for `constant_values`?

Something like:
```python
nodata = self.encoded_nodata if self.encoded_nodata is not None else self.nodata
if nodata is None:
    nodata = np.nan
```

Reply:
Sounds good, or even passing `constant_values` to `.pad_box(minx, miny, maxx, maxy, constant_values=None)` (and `pad_xy`), then in `pad_xy`:
```python
if constant_values is None:
    constant_values = self.encoded_nodata if self.encoded_nodata is not None else self.nodata
if constant_values is None:
    constant_values = np.nan
```
i.e. if the user knows what he wants, he should pass it explicitly, and otherwise we use the value most likely to represent no data. What do you think?

Reply:
That makes sense to me.